lib/publiccloud/instance.pm

```diff
@@ -79,6 +79,8 @@ sub run_ssh_command {
     } elsif ($rc_only) {
         # Increase the hard timeout for script_run, otherwise our 'timeout $args{timeout} ...' has no effect
         $args{timeout} += 2;
+        # Pipe both the standard and error output serial for debug purposes
+        $ssh_cmd .= " 2>&1 | tee $serialdev";
```

Comment:
looks good

Reply:
In fact no it's not good: the purpose of `rc_only` is to return the "return code" of `$ssh_cmd`, right? In that case adding `| tee $serialdev` is bad, as the last return code of `$ssh_cmd` will be the one from the `tee` command, and in many case it's always 0.

Here an example of a failing test: http://1b210.qa.suse.de/tests/8007#step/sles4sap/420, it should returned RC=102 but in fact it returned 0.

I will do a PR to fix this.